### PR TITLE
allowing setting node variables from test

### PR DESCRIPTION
**Pull request by nshidlin**
This PR adds:

- request argument to nodes fixture in order to set vm creation parameters at the test level
- modifies attach_disk fixture to support multiple disks

### CODE REVIEW

File: discovery-infra/tests/base_test.py [hunk 1]
```diff
@@ -228,3 +239,12 @@ def update_oc_config(self, nodes, cluster):
...
                                api_vip=api_vip)
+
+    @staticmethod
+    def get_insufficeint_disk_size_for_nodes(api_client):
+        min_host_requirements = api_client.get_host_requirements()
+        min_disk = min_host_requirements.worker.disk_size_gb
+        min_disk_bytes = min_disk * pow(1000, 3)
+        insufficient_disk_bytes = math.floor(min_disk_bytes * 0.90)
+
+        return {'worker_disk': insufficient_disk_bytes, 'master_disk': insufficient_disk_bytes}
```
Comment (lalon4) [inline discovery-infra/tests/base_test.py:250]: why we need to return same value twice? cant we just return node_disk?

Comment (nshidlin) [reply discovery-infra/tests/base_test.py:250]: `worker_disk` and `master_disk` are the names of the terrafrom variables. This makes it easier to pass to override_vm_parameters

Comment (lalon4) [reply discovery-infra/tests/base_test.py:250]: think its better to have this method in test level. WDYT?

Comment (nshidlin) [reply discovery-infra/tests/base_test.py]: done

File: discovery-infra/tests/base_test.py [hunk 2]
```diff
@@ -18,18 +20,27 @@
 from tests.conftest import env_variables, qe_env
 from download_logs import download_logs
 
-
 class BaseTest:
+    @staticmethod
+    def override_vm_parameters(**kwargs):
```
Comment (tsorya) [inline discovery-infra/tests/base_test.py:25]: where it is used?

Comment (nshidlin) [reply discovery-infra/tests/base_test.py:25]: in the tests

Comment (nshidlin) [reply discovery-infra/tests/base_test.py]: moved the calling of this into `nodes` fixture

**Pull request merged**